### PR TITLE
add graphql server / endpoint

### PR DESCRIPTION
**Pull request by willscott**
* generation of graphql schema over ipld types
* secondary servlet at separate port if specified as flag
* returns list of 'getall' tasks as initial available query.

### CODE REVIEW

File: controller/graphql/server.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+				Context:       ctx,
+				Schema:        schema,
+				RequestString: r.URL.Query().Get("query"),
+			})
+		}
+
+		if len(result.Errors) > 0 {
+			log.Printf("Query had errors: %s, %v", r.URL.Query().Get("query"), result.Errors)
+		}
+		json.NewEncoder(w).Encode(result)
```
Comment (mvdan) [inline controller/graphql/server.go:92]: check this error? at least we could log it and/or return a generic error string

File: controller/graphql/server.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+	mux := http.NewServeMux()
+	mux.Handle("/graphql", CorsMiddleware(func(w http.ResponseWriter, r *http.Request) {
+		var result *graphql.Result
+		ctx := r.Context()
+
+		if r.Method == "POST" && r.Header.Get("Content-Type") == "application/json" {
+			var p postData
+			if err := json.NewDecoder(r.Body).Decode(&p); err != nil {
+				http.Error(w, err.Error(), 500)
```
Comment (mvdan) [inline controller/graphql/server.go:58]: seems to me like this should be a BadRequest back to the user, not a ServerInternalError

File: controller/graphql/server.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		} else {
+			result = graphql.Do(graphql.Params{
+				Context:       ctx,
+				Schema:        schema,
+				RequestString: r.URL.Query().Get("query"),
+			})
+		}
+
+		if len(result.Errors) > 0 {
+			log.Printf("Query had errors: %s, %v", r.URL.Query().Get("query"), result.Errors)
```
Comment (mvdan) [inline controller/graphql/server.go:90]: I guess the user would get these too?

**Review by mvdan — APPROVED** [review]

File: controller/graphql/server.go (hunk 3 above)
Comment (willscott) [reply controller/graphql/server.go:90]: yep. they're part of the result object returned

**Pull request merged**